### PR TITLE
Implement eth_feeHistory RPC endpoint

### PR DESCRIPTION
**Pull request by LogvinovLeon**
Adds `eth_feeHistory` to expose the rollup's EIP-1559 base fee history, enabling wallets to query gas prices and set appropriate `max_fee_per_gas` values.

## Implementation

- New module: `sov-evm/src/rpc/fee_history.rs`
- Fetches rollup gas prices from ChainState, converts to EVM format (dimension 0)
- Returns N+1 base fees (N blocks + next) and N gas used ratios
- Caps at 1024 blocks, errors on 0 blocks
- Priority fees return zeros (preferred sequencer model)

## Testing

6 test cases covering basic usage, edge cases, and validation.

```bash
cargo nextest run fee_history
```

# Description

*Summary of the changes...*

- [ ] I have updated `CHANGELOG.md` with a new entry if my PR makes any breaking changes or fixes a bug. If my PR removes a feature or changes its behavior, I provide help for users on how to migrate to the new behavior.
- [ ] I have carefully reviewed all my `Cargo.toml` changes before opening the PRs. (Are all new dependencies necessary? Is any module dependency leaked into the full-node (hint: it shouldn't)?)

## Linked Issues
- Fixes # (issue, if applicable)
- Related to # (issue) 

## Testing
Describe how these changes were tested. If you've added new features, have you added unit tests?

## Docs
Describe where this code is documented. If it changes a documented interface, have the docs been updated?

Rendered docs are available at `https://sovlabs-ci-rustdoc-artifacts.us-east-1.amazonaws.com/<BRANCH_NAME>/index.html`

### CODE REVIEW

File: crates/module-system/module-implementations/sov-evm/src/rpc/handlers.rs [hunk 1]
```diff
@@ -187,11 +177,32 @@ where
...
-        Ok(EMPTY_FEE_HISTORY)
+    pub fn fee_history(
+        &self,
+        block_count: U64,
+        newest_block: BlockNumberOrTag,
+        reward_percentiles: Option<Vec<f64>>,
+        state: &mut ApiStateAccessor<S>,
+    ) -> RpcResult<FeeHistory> {
+        trace!(
+            block_count = block_count.to::<u64>(),
```
Comment (bkolad) [inline crates/module-system/module-implementations/sov-evm/src/rpc/handlers.rs]: block_count.to::<u64>() is repeated 2x. Here and in line 201. Let's extract it.

**Review by bkolad — APPROVED** [review]
